cpp/core/memory/Reclaimable.h

```diff
@@ -21,11 +21,11 @@
 
 namespace gluten {
 
-class Evictable {
+class Reclaimable {
```

Comment:
It's not a big deal but does seem like we have mixed naming of `reclaim`, `evict` and `spill` in shuffle.

I recall that previously `evict` could be considered an abstraction of local writer's `spill`. So if we introduce `reclaim`, how should it stand for?